client/src/pages/Messages/ChatBox.tsx

```diff
@@ -0,0 +1,240 @@
+  const replyAreaRef = useRef<HTMLInputElement>(null);
+
+  const { updateSnackBarMessage } = useSnackBar();
+  const { socket } = useSocket();
+
+  const handleOnKeyPress = (event: React.KeyboardEvent<HTMLInputElement>) => {
+    if (event.key === 'Enter' && event.shiftKey) {
+      return;
+    } else if (event.key === 'Enter') {
+      handleSendMessage();
+    }
+  };
+
+  const handleSendMessage = () => {
+    const AsyncSendMessage = async () => {
+      if (
+        currentConversation &&
+        currentRecipient &&
+        currentConversation.conversationId &&
+        currentRecipient._id &&
+        socket
+      ) {
+        const conversationId = currentConversation.conversationId;
+        const recipientId = currentRecipient._id;
+
+        const response = await sendMessage(conversationId, messageText, recipientId);
+        if (response.error) {
+          updateSnackBarMessage(response.error);
+        } else if (response.message) {
+          const emitMessage = {
+            recipientId,
+            ...response.message,
+          };
+          socket.emit('sendMessage', emitMessage);
+          setMessageText('');
+
+          if (replyAreaRef.current) {
+            replyAreaRef.current.value = '';
+          }
+
+          const setCurrentConversation = async () => {
+            const response = await getOneConversation(conversationId);
+
+            if (response.error) {
+              updateSnackBarMessage(response.error);
+            } else if (response.conversationContent) {
+              updateCurrentConversation(response.conversationContent);
+            }
+          };
+
+          setCurrentConversation();
+        }
+      }
+    };
+
+    AsyncSendMessage();
+  };
+
+  const handleSendTextChange = (e: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    const messageTxt = e.target.value;
+    setMessageText(messageTxt);
+  };
+
+  useEffect(() => {
+    if (socket) {
+      socket.on('getMessage', (message) => {
+        setArrivalMessage(message);
+      });
+    }
+  }, [socket]);
+
+  useEffect(() => {
+    if (arrivalMessage) setMessages((prev) => [...prev!, arrivalMessage]);
+  }, [arrivalMessage]);
+
+  useEffect(() => {
+    let active = true;
+
+    const getRecipient = (): Profile | undefined => {
+      const currentConversationId = currentConversation?.conversationId;
+      const currentConversationWithMembers = conversations?.filter(
+        (conversation) => conversation._id === currentConversationId,
+      );
+      const conversationMembers =
+        currentConversationWithMembers &&
+        currentConversationWithMembers[0] &&
+        currentConversationWithMembers[0].members;
+      const recipient = conversationMembers?.filter((memebr) => memebr._id != profileId)[0];
+      return recipient;
+    };
+    currentConversation && setCurrentRecipient(getRecipient());
+    currentConversation && currentConversation.messages && setMessages(currentConversation?.messages);
```

Comment:
You could shorten this line with optional chaining:
`currentConversation?.messages && setMessages(currentConversation?.messages);`

Reply:
done ✔️